pandas-stubs/core/frame.pyi

```diff
@@ -229,6 +235,11 @@ class _LocIndexerFrame(_LocIndexer):
         value: Scalar | NAType | NaTType | ArrayLike | Series | list | None,
     ) -> None: ...
 
+_TT = TypeVar("TT", bound=Literal[True, False])
+
+class DataFrameGroupByGen(DataFrameGroupBy[ByT], Generic[ByT, _TT]): ...
```

Comment:
@Dr-Irv these seem to be more what you had originally in mind, please confirm if that is what you were thinking

Reply:
No, I was thinking of doing this:
```python
class DataFrameGroupBy(GroupBy[DataFrame], Generic[ByT, _TT]):
```
No need to introduce a new class